### PR TITLE
Implementation of SetApprovalForAll system contract

### LINKED ISSUE #8590: APPROVE_OPERATOR (SetApprovalForAllPrecompile)

### PR DESCRIPTION
**Pull request by MrValioBg**
**Description**:
Implementation for the SetApprovalForAll operation.

This PR includes the implementation & tests both for the ERC and classical selector call.

**Related issue(s)**:

Fixes #8590 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java [hunk 1]
```diff
@@ -115,8 +115,7 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        final var txn = context.body();
-        final var payer = txn.transactionIDOrThrow().accountIDOrThrow();
+        final var payer = context.payer();
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java:118]: 🙌

**Review by tinker-michaelj — DISMISSED** [review]
LGTM

**Comment by tinker-michaelj** [comment]
Re-running checks on this PR, seems they either canceled or hit flaky tests. Let's see. 🤞

**Review by stoyanov-st — DISMISSED** [review]
LGTM

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/setapproval/SetApprovalForAllTranslator.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import com.hedera.node.app.spi.workflows.record.SingleTransactionRecordBuilder;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Arrays;
+import javax.inject.Inject;
+
+/**
+ * Translates setApprovalForAll call to the HTS system contract. There are no special cases for
+ * these calls, so the returned {@link HtsCall} is simply an instance of {@link DispatchForResponseCodeHtsCall}.
+ */
+public class SetApprovalForAllTranslator extends AbstractHtsCallTranslator {
```
[the remark this answers is not in the document]
Comment (MrValioBg) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/setapproval/SetApprovalForAllTranslator.java:35]: Thanks, done :)

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by Nana-EC — APPROVED** [review]

**Review by povolev15 — APPROVED** [review]

**Pull request merged**